domain/domain.go

```diff
@@ -526,7 +526,7 @@ func NewDomain(store kv.Storage, ddlLease time.Duration, statsLease time.Duratio
 		store:           store,
 		SchemaValidator: NewSchemaValidator(ddlLease),
 		exit:            make(chan struct{}),
-		sysSessionPool:  pools.NewResourcePool(factory, capacity, capacity, resourceIdleTimeout),
+		sysSessionPool:  newSessionPool(capacity, factory),
```

Comment:
do we have maxIdle(or max pool) and too many idle session that make memory leak or oom question?

Reply:
The max size of the sessionPool is set to the `capacity`.
Although after the pool object size reach the capacity, it doesn't block on allocating.
@lysu

Reply:
memory leak or oom depends on how the caller use the pooled session.
Restricted SQLs are write by us, so I think there should not be any problem.